hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,374 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2015 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+# INFORMATION ABOUT THE EMD FORMAT CAN BE FOUND HERE:
+# http://emdatasets.lbl.gov/
+
+
+import re
+import h5py
+
+import logging
+
+
+# Plugin characteristics
+# ----------------------
+format_name = 'Electron Microscopy Data (EMD)'
+description = 'Read data from Berkleys EMD files.'
+full_support = True  # Hopefully?
+# Recognised file extension
+file_extensions = ('emd', 'EMD')
+default_extension = 0
+# Writing features
+writes = True
+EMD_VERSION = '0.2'
+# ----------------------
+
+
+class EMD(object):
+
+    '''Class for storing electron microscopy datasets.
+
+    The :class:`~.EMD` class can hold an arbitrary amount of datasets in the `signals` dictionary.
+    These are saved as HyperSpy :class:`~hyperspy.signal.Signal` instances. Global metadata
+    are saved in four dictionaries (`user`, `microscope`, `sample`, `comments`). To print
+    relevant information about the EMD instance use the :func:`~.print_info` function. EMD
+    instances can be loaded from and saved to emd-files, an hdf5 standard developed at Lawrence
+    Berkeley National Lab (http://emdatasets.lbl.gov/).
+
+    Attributes
+    ----------
+    signals: dictionary
+        Dictionary which contains all datasets as :class:`~hyperspy.signal.Signal` instances.
+    user: dictionary
+        Dictionary which contains user related metadata.
+    microscope: dictionary
+        Dictionary which contains microscope related metadata.
+    sample: dictionary
+        Dictionary which contains sample related metadata.
+    comments: dictionary
+        Dictionary which contains additional commentary metadata.
+
+    '''
+
+    _log = logging.getLogger(__name__)
+
+    def __init__(self, signals={}, user={}, microscope={}, sample={}, comments={}):

```

Comment:
We encourage not to use new/empty dicts as default values. Instead you should write, e.g.

```python
def __init__(self, signals=None, ): # same for others if appropriate
    if signals is None:
        signals = {}
    # other default checks / code follows
```